Sources/SecureXPC/Server/XPCMachServer.swift

```diff
@@ -181,32 +181,31 @@ extension XPCMachServer {
         }
 
         // Generate client requirements from info property list's SMAuthorizedClients
-        var clientRequirements = [SecRequirement]()
         let infoData = try readEmbeddedPropertyList(sectionName: "__info_plist")
-        let infoPropertyList = try PropertyListSerialization.propertyList(
-            from: infoData,
-            options: .mutableContainersAndLeaves,
-            format: nil) as? NSDictionary
-        if let authorizedClients = infoPropertyList?["SMAuthorizedClients"] as? [String] {
-            for client in authorizedClients {
-                var requirement: SecRequirement?
-                if SecRequirementCreateWithString(client as CFString, SecCSFlags(), &requirement) == errSecSuccess,
-                   let requirement = requirement {
-                    clientRequirements.append(requirement)
-                } else {
-                    throw XPCError.misconfiguredServer(description: "Invalid SMAuthorizedClients requirement: " +
-                                                                    "\(client)")
-                }
-            }
-        } else {
+        let infoPropertyList = try PropertyListSerialization.propertyList(from: infoData, format: nil) as? NSDictionary
+        guard let authorizedClients = infoPropertyList?["SMAuthorizedClients"] as? [String] else {
             throw XPCError.misconfiguredServer(description: "Info property list missing SMAuthorizedClients key")
         }
-        if clientRequirements.isEmpty {
+        
+        var clientRequirement: XPCClientRequirement? = nil
+        for client in authorizedClients {
+            var requirement: SecRequirement?
+            guard SecRequirementCreateWithString(client as CFString, SecCSFlags(), &requirement) == errSecSuccess,
+                  let requirement = requirement else {
+                throw XPCError.misconfiguredServer(description: "Invalid SMAuthorizedClients requirement: \(client)")
+            }
+            
+            if let currentRequirement = clientRequirement {
+                clientRequirement = .or(currentRequirement, .secRequirement(requirement))
+            } else {
+                clientRequirement = .secRequirement(requirement)
+            }
```

Comment:
This is essentially building up a linked list. Would an array be preferable? It'd be simpler to construct (just a simple `compactMap`), simpler to check `contains(where: $0.acceptsMessage(...))`, and might be nicer on allocations/memory.

The math nerd in me is compelled to mention that this is called "existential quantification" ("any"), and the opposite is "universal quantification" ("all")